frontend/console/src/features/modules/decls/TypeEl.tsx

```diff
@@ -0,0 +1,66 @@
+import type { Optional, Ref, Array as SchArray, Map as SchMap, Type } from '../../../protos/xyz/block/ftl/v1/schema/schema_pb'
+import { DeclLink } from './DeclLink'
+
+// TypeParams ironically is built to work with the `Type` message, not the
+// `TypeParameter` message, which just has a simple string param name without
+// any higher level type information.
+const TypeParams = ({ types }: { types?: (Type | undefined)[] }) => {
+  if (!types) {
+    return []
```

Comment:
I feel like these should return `null` or `undefined` instead of `[]` I keep getting hung up on the fact that things return either an empty array or a react component. Maybe we can pick a pattern and just make sure we use that everywhere?

Reply:
Or even better, we should just `return` with nothing, which should be equivalent and more readable. 
```ts
if (!types) {
  return
}
```
or 
```ts
if (!types) return
```

Reply:
In the past, I've noticed both null and undefined ended up being treated as real children in the React component tree when you returned anything other than `[]`. I'll see if that's still an issue.

Reply:
They finally fixed it!! Yay! Cool I'll switch to returning nothing/undefined